zipkin-lens/src/components/App/App.test.js

```diff
@@ -37,34 +38,48 @@ describe('<App />', () => {
     mount.cleanUp();
   });
 
-  it('should update document title', () => {
+  it('should update document title when mounted', () => {
+    // Since it is not possible to verify that document.title has
+    // been changed using shallow rendering, use mount.
     mount(<App />);
     expect(document.title).toBe('Zipkin');
   });
 
-  it('should render providers', () => {
+  it('should provide React Context for MuiPickers', () => {
     const wrapper = shallow(<App />);
     expect(wrapper.find(MuiPickersUtilsProvider).length).toBe(1);
+    expect(wrapper.find(MuiPickersUtilsProvider).props().utils).toEqual(MomentUtils);
+  });
+
+  it('should provide only one theme', () => {
+    const wrapper = shallow(<App />);
     expect(wrapper.find(ThemeProvider).length).toBe(1);
     expect(wrapper.find(ThemeProvider).props().theme).toEqual(theme);
-    expect(wrapper.find(Provider).length).toBe(1);
   });
 
-  it('should render Router as a parent of Layout', () => {
+  it('should provide redux store', () => {
     const wrapper = shallow(<App />);
-    expect(wrapper.find(Layout).parent().type()).toEqual(BrowserRouter);
+    expect(wrapper.find(Provider).length).toBe(1);
   });
 
-  it('should render Layout', () => {
+  it('should share Layout among all page components', () => {
     const wrapper = shallow(<App />);
-    expect(wrapper.find(Layout).length).toBe(1);
+    const layout = wrapper.find(Layout);
+    expect(layout.length).toBe(1);
+    // In order to share Layout, the Route components must be defined under Layout.
+    expect(layout.find(Route).length).toBe(2);
+    // Since the Layout component is wrapped with withRouter, it needs to be defined
+    // under BrowserRouter.
+    expect(layout.parent().type()).toEqual(BrowserRouter);
   });
 
-  it('should render Route', () => {
+  it('should render 2 Route', () => {
     const wrapper = shallow(<App />);
     const routes = wrapper.find(Route);
+    // Check routes that use a single trace as input.
```

Comment:
I think this should say `// Check routes that use the same global search bar` or similar right? these are about multiple traces

Reply:
Oops!